app/javascript/dropdown.js

```diff
@@ -0,0 +1,37 @@
+$(document).ready(function() {
+  $(document).on('click', '.btn-dropdown', function() {
+    $('.dropdown-menu').toggle();
+  });
+
+  $(document).on('click', '.dropdown-item', function() {
+    $('.dropdown-menu').hide();
+  });
+
+  $(document).on('click', '.dropdown-item', function() {
+    const self = $(this);
+    const sortBy = self.data('sort-by');
+    $.ajax ({
+      url: '/posts',
+      method: 'GET',
+      data: {
+        sort_by: sortBy,
+      },
+      dataType: 'json',
+      success: function(data) {
+        $('.posts-container').html(data.attachment_partial);
+        applyButtonStyling();
+        
+        $('.btn-dropdown').text(sortBy);
+      }
+    });
+  });
```

Comment:
This two onClick functions are triggered at the same time.

I think you could add the  `$('.dropdown-menu').hide();` at the second one and avoid defining two functions.